### PR TITLE
[ESLint Hackathon] Fix lint warnings in `react-components` module.

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/12726

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/react-components/src/components/accordion/segmented-accordion/segmented-accordion-title.tsx [hunk 1]
```diff
@@ -230,11 +230,11 @@ export const SegmentedAccordionTitle: FunctionComponent<SegmentedAccordionTitleP
                                         transparent
                                         verticalAlign="middle"
                                         icon={
-                                            <Icon
+                                            (<Icon
```
Comment (kayathiri4) [inline modules/react-components/src/components/accordion/segmented-accordion/segmented-accordion-title.tsx:233]: Can we bring `<Icon..` to the next line?

File: modules/react-components/src/components/danger-zone/danger-zone.tsx [hunk 2]
```diff
@@ -107,18 +107,18 @@ export const DangerZone: FunctionComponent<DangerZoneProps> = (
             </Header>
             {
                 toggle ?
-                    <Checkbox
+                    (<Checkbox
```
Comment (kayathiri4) [inline modules/react-components/src/components/danger-zone/danger-zone.tsx:110]: Shall we format here?

File: modules/react-components/src/components/input/file-upload.tsx [hunk 3]
```diff
@@ -303,10 +307,13 @@ export const FileUpload: FunctionComponent<FileUploadPropsInterface> = (
...
-                                    event.preventDefault();
-                                    fileUpload.current.click();
-                                } }>
+                                <Button
+                                    basic
+                                    primary
+                                    onClick={ (event) => {
+                                        event.preventDefault();
+                                        fileUpload.current.click();
+                                    } }>
```
Comment (kayathiri4) [inline modules/react-components/src/components/input/file-upload.tsx:316]: Shall we change as 
```
<Button
     onClick={ () => {                      
     } }
>
```

File: modules/react-components/src/components/input/file-upload.tsx [hunk 4]
```diff
@@ -315,14 +322,17 @@ export const FileUpload: FunctionComponent<FileUploadPropsInterface> = (
...
-                                } }/>
+                                <Icon
+                                    name="trash alternate"
+                                    link
+                                    onClick={ () => {
+                                        setFile(null);
+                                        setContent("");
+                                        setEncodedContent("");
+                                        setFileError(false);
+                                    } }/>
```
Comment (kayathiri4) [inline modules/react-components/src/components/input/file-upload.tsx:335]: Shall we bring `/>` to next line

File: modules/react-components/src/components/input/inline-edit-field.tsx [hunk 5]
```diff
@@ -138,13 +140,13 @@ export const InlineEditInput: FunctionComponent<InlineEditInputPropsInterface> =
                         />
                     </Grid.Column>
                 </Grid.Row>
-            </Grid>
-        :
-            <Grid columns="equal" verticalAlign="middle">
+            </Grid>)
+            :
+            (<Grid columns="equal" verticalAlign="middle">
```
Comment (kayathiri4) [inline modules/react-components/src/components/input/inline-edit-field.tsx:145]: Shall we do formatting here?

File: modules/react-components/src/components/list/resource-list/resource-list-item.tsx [hunk 6]
```diff
@@ -126,13 +126,14 @@ export const ResourceListItem: FunctionComponent<ResourceListItemPropsInterface>
...
-                            : 2
-                }>
+                <Grid.Row
+                    columns={
+                        metaContent instanceof Array
+                            ? (metaContent.length + 2) as StrictGridRowProps[ "columns" ]
+                            : (itemHeader || itemDescription)
+                                ? 3
+                                : 2
+                    }>
```
Comment (kayathiri4) [inline modules/react-components/src/components/list/resource-list/resource-list-item.tsx:136]: Shall we bring `>` to next line?

File: modules/react-components/src/components/danger-zone/danger-zone.tsx (hunk 2 above)
Comment (brionmario) [reply modules/react-components/src/components/danger-zone/danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2712/commits/822ffa6c9b9d9f29aad197efcd574ef27c3e7440 & https://github.com/wso2/identity-apps/pull/2712/commits/f04b53ae80f3c688374854052c552a5f698db22f.

File: modules/react-components/src/components/list/resource-list/resource-list-item.tsx (hunk 6 above)
Comment (brionmario) [reply modules/react-components/src/components/list/resource-list/resource-list-item.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2712/commits/822ffa6c9b9d9f29aad197efcd574ef27c3e7440 & https://github.com/wso2/identity-apps/pull/2712/commits/f04b53ae80f3c688374854052c552a5f698db22f.

File: modules/react-components/src/components/input/file-upload.tsx (hunk 3 above)
Comment (brionmario) [reply modules/react-components/src/components/input/file-upload.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2712/commits/822ffa6c9b9d9f29aad197efcd574ef27c3e7440 & https://github.com/wso2/identity-apps/pull/2712/commits/f04b53ae80f3c688374854052c552a5f698db22f.

**Review by thivi — APPROVED** [review]

**Pull request merged**